WordPress/Classes/ViewRelated/Post/Prepublishing/PrepublishingViewController+JetpackSocial.swift

```diff
@@ -197,9 +169,72 @@ private extension PrepublishingViewController {
         }
     }
 
+    // MARK: - Model Creation
+
+    func makeAutoSharingModel() -> PrepublishingAutoSharingModel {
+        return coreDataStack.performQuery { [postObjectID = post.objectID] context in
+            guard let post = (try? context.existingObject(with: postObjectID)) as? Post,
+                  let connections = post.blog.sortedConnections as? [PublicizeConnection],
+                  let supportedServices = try? PublicizeService.allSupportedServices(in: context) else {
+                return .init(services: [], message: String(), sharingLimit: nil)
+            }
+
+            // first, build a dictionary to categorize the connections.
+            var connectionsMap = [PublicizeService.ServiceName: [PublicizeConnection]]()
+            connections.filter { !$0.requiresUserAction() }.forEach { connection in
+                let serviceName = PublicizeService.ServiceName(rawValue: connection.service) ?? .unknown
+                var serviceConnections = connectionsMap[serviceName] ?? []
+                serviceConnections.append(connection)
+                connectionsMap[serviceName] = serviceConnections
+            }
+
+            // then, transform [PublicizeService] to [PrepublishingAutoSharingModel.Service].
+            let modelServices = supportedServices.compactMap { service -> PrepublishingAutoSharingModel.Service? in
+                // skip services without connections.
+                guard let serviceConnections = connectionsMap[service.name],
+                      !serviceConnections.isEmpty else {
+                    return nil
+                }
+
+                return PrepublishingAutoSharingModel.Service(
+                    name: service.name,
+                    connections: serviceConnections.map {
+                        .init(account: $0.externalDisplay,
+                              keyringID: $0.keyringConnectionID.intValue,
+                              enabled: !post.publicizeConnectionDisabledForKeyringID($0.keyringConnectionID))
+                    }
+                )
+            }
+
+            return .init(services: modelServices,
+                         message: post.publicizeMessage ?? post.titleForDisplay(),
+                         sharingLimit: post.blog.sharingLimit)
+        }
+    }
+
     // MARK: - Constants
 
     enum Constants {
         static let noConnectionKey = "prepublishing-social-no-connection-view-hidden"
     }
 }
+
+// MARK: - Auto Sharing Model
+
+/// A value-type representation of `PublicizeService` for the current blog that's simplified for the auto-sharing flow.
+struct PrepublishingAutoSharingModel {
```

Comment:
I've converted the previous `PrepublishingAutoSharingViewModel` that's exclusively used in `PrepublishingAutoSharingView` to `PrepublishingAutoSharingModel`, and moved it to the `PrepublishingViewController+JetpackSocial`. The reason is that this model is shared with the social accounts view.